core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java

```diff
@@ -2042,6 +2042,27 @@ public Node visitExtract(SqlBaseParser.ExtractContext context)
         return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), field);
     }
 
+    @Override
+    public Node visitTrim(SqlBaseParser.TrimContext context)
+    {
+        String functionName = getTrimFunctionName((Token) context.trimsSpecification().getChild(0).getPayload());
+        List<Expression> arguments = Lists.reverse(visit(context.valueExpression(), Expression.class));
+        return new FunctionCall(getLocation(context), QualifiedName.of(functionName), arguments);
+    }
+
+    private static String getTrimFunctionName(Token token)
```

Comment:
This differentiation should go to the Analyzer, where you have to resolve the function.